pkg/operator/etcdmemberipmigrator/etcdmemberipmigrator.go

```diff
@@ -152,6 +156,51 @@ func (c *EtcdMemberIPMigrator) reconcileMembers() error {
 	return nil
 }
 
+// getRequiredPeerListFromDNS finds the member IP using a reverse DNS lookup.
+// This approach is very heavy with dns requests but this should almost never
+// happen and this allows us to re-use the code that we have already for DNS.
+func (c *EtcdMemberIPMigrator) getRequiredPeerListFromDNS(member *etcdserverpb.Member) ([]string, error) {
+	etcdInfo, err := c.getEtcdInfo()
+	if err != nil {
+		return nil, err
+	}
+	return getRequiredPeerList(etcdInfo, member.PeerURLs)
+}
+
+// getRequiredPeerListFromFallback tries to map a member to an IP address by
+// exploiting naming conventions. For example, on Azure IPI, we know the member
+// names will be a function of the node name, and so we can find the node based
+// on the member name and use the node's IP without a DNS query.
+func (c *EtcdMemberIPMigrator) getRequiredPeerListFromFallback(member *etcdserverpb.Member) ([]string, error) {
+	infra, err := c.infrastructureLister.Get("cluster")
+	if err != nil {
+		return nil, err
+	}
+	if infra.Status.PlatformStatus == nil {
+		return nil, fmt.Errorf("no reliable platform metadata is available")
+	}
+
+	switch infra.Status.PlatformStatus.Type {
+	case configv1.AzurePlatformType:
+		network, err := c.networkLister.Get("cluster")
+		if err != nil {
+			return nil, err
+		}
+		nodeName := strings.ReplaceAll(member.Name, "etcd-member-", "")
```

Comment:
Are we 100% certain this will always work even with Azure UPI? Why not use member list here to get node IP?

Reply:
No, but we have no data about this fallback being useful except in the following conditions:

1. Azure IPI
2. DNS forwarding enabled _for the cluster domain_
3. The upstream is misconfigured in that it excludes the etcd records or fails to delegate back to the OpenShift-defined zones containing the etcd records

So I don't really know it matters whether it breaks outside those conditions. This PR isn't intended to address every conceivable DNS forwarding problem. The answer in many cases may in fact be to ask the user to fix their nameservers.

Reply:
OK as long as are not introducing a new failure case but looking this over the worst case is retry.